packages/ckeditor5-ui/src/list/listitemgroupview.ts

```diff
@@ -0,0 +1,120 @@
+/**
+ * @license Copyright (c) 2003-2023, CKSource Holding sp. z o.o. All rights reserved.
+ * For licensing, see LICENSE.md or https://ckeditor.com/legal/ckeditor-oss-license
+ */
+
+/**
+ * @module ui/list/listitemgroupview
+ */
+
+import View from '../view';
+import type ViewCollection from '../viewcollection';
+import ListView from './listview';
+
+import { uid, type Locale } from '@ckeditor/ckeditor5-utils';
+
+/**
+ * The list item group view class.
+ */
+export default class ListItemGroupView extends View {
+	/**
+	 * The visible label of the group.
+	 *
+	 * @observable
+	 * @default ''
+	 */
+	declare public label: string;
+
+	/**
+	 * Collection of the child list items inside this group.
+	 */
+	public readonly items: ListView[ 'items' ];
+
+	/**
+	 * Collection of the child elements of the group.
+	 */
+	public readonly children: ViewCollection;
+
+	/**
+	 * Controls whether the item view is visible. Visible by default, list items are hidden
+	 * using a CSS class.
+	 *
+	 * @observable
+	 * @default true
+	 */
+	declare public isVisible: boolean;
+
+	/**
+	 * @inheritDoc
+	 */
+	constructor( locale?: Locale ) {
+		super( locale );
+
+		const bind = this.bindTemplate;
+		const groupLabelId = `ck-editor__label_${ uid() }`;
+		const nestedList = new ListView( locale );
+
+		this.children = this.createCollection();
+		this.children.addMany( [ this._createLabel( groupLabelId ), nestedList ] );
+
+		this.set( {
+			label: '',
+			isVisible: true
+		} );
+
+		nestedList.set( {
+			role: 'group',
+			ariaLabelledBy: groupLabelId
+		} );
+
+		// Disable focus tracking and accessible navigation in the child list.
+		nestedList.focusTracker.destroy();
+		nestedList.keystrokes.destroy();
+
+		this.items = nestedList.items;
+
+		this.setTemplate( {
+			tag: 'li',
+
+			attributes: {
+				role: 'presentation',
+				class: [
+					'ck',
+					'ck-list__group',
+					bind.if( 'isVisible', 'ck-hidden', value => !value )
+				]
+			},
+
+			children: this.children
+		} );
+	}
+
+	/**
+	 * Creates a label for the group.
+	 */
+	private _createLabel( groupLabelId: string ): View {
```

Comment:
I think a labelless group contradicts the idea of a grouping. It does not make sense visually. It does not make sense accessibility-wise either. 

If it wasn't for the fact that we don't provide values via constructors but use observables (I mentioned this in a previous comment that we should really figure it out one day), I'd either block the possibility of passing an empty string or even throw/warn if that ever happened.

Reply:
Visually it makes a bit of sense because the groups can be separated by, well, some visual element, like a border. Do you see maybe a use for `ListItemGroupView` in UIs like nested dropdowns or application menu? Or will it be just an array of list items anyway?

I was more afraid about a11y issues.

But maybe if you want to have a group, you should always have a name for it (a11y, something else?) and then if you do not want to display it, you can have a proper template or even use CSS.

I was coming just from the visual POV -- in case if someone wants to have the options seprated but not to include a label. Well, similar thing as in toolbar, you also create groups but don't name them.

Reply:
There's a `ListItemSeparator` component in out codebase for that purpose.